src/plots/cartesian/set_convert.js

```diff
@@ -182,8 +182,9 @@ module.exports = function setConvert(ax) {
             // encounters them, ie all the categories from the
             // first data set, then all the ones from the second
             // that aren't in the first etc.
-            // TODO: sorting options - do the sorting
-            // progressively here as we insert?
+            // it is assumed that this function is being invoked in the
+            // already sorted category order; otherwise there would be
+            // a disconnect between the array and the index returned
```

Comment:
@etpinard I'll look into it more closely today, but on an initial look, the suggested place isn't as trivial (to me) and I'd put it slightly after this point, because 

- setAutoType bails if `ax.type!=='-'` and at the only place it's called from, there is an identical (redundant) check
- the loop in question depends on `isBoxWithoutPositionCoords` returning true
- the inside of the loop (now) is solely dedicated to building up the boxPositions
- if axis type `'category`' isn't explicitly specified, it won't be known until the subsequent call to `autoType()` at the end

Therefore I'm planning to put the logic _after_ having returned from setAutoType; the earliest point seems to be just before the call to `setConvert`. 

This way, we wouldn't reuse the suggested loop, but the loop over the traces probably doesn't have a measurable performance impact anyway (I'm assuming there usually aren't thousands or 10ks of trace lines).

I'll do some more tests to ensure I'm not overlooking something; just wanted to share current thinking.